File: FDFrontend/src/app/model/menu.ts
import { Category } from "./category";
import { Restaurant } from "./restaurant";

export class Menu{
    public id: number | null;
    public name: string | null;
    public description: string | null;
    public price: number | null;
    public category: Category;

    constructor(){
        this.id = null;
        this.name = null;
        this.description = null;
        this.price = null;
        this.category = new Category();
    }
}

File: FDBackend/src/main/java/FDBackend/Entities/Menu.java
package FDBackend.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "menus")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Menu {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    private String name;

    @Column
    private String description;

    @Column
    private double price;

    @ManyToOne
    @JoinColumn(name = "restaurantId")
    @JsonIgnore
    private Restaurant restaurant;

    @ManyToOne
    @JoinColumn(name = "categoryId")
    private Category category;

    @OneToMany(mappedBy = "menu")
    @JsonIgnore
    private List<FoodCart> foodCarts = new ArrayList<>();
}
